arm/proofs/decode.ml

```diff
@@ -592,6 +715,20 @@ let PURE_DECODE_CONV =
         | _ -> failwith "eval_opt")
     | ty -> failwith ("eval_opt " ^ string_of_type ty) in
 
+  (* Evaluates term t in a continuation-passing style. The continuation
+     'F(thm, binding)' takes (1) thm: a rewrite rule that describes the
+     equality `e = v` where `e` was the previous expression and `v` is the
+     result of evaluation (2) binding: a list of free variables and
+     their values.
+     To understand this function further, you can start with the "LET"
+     case which is HOL Light's let binding `let x = e1 in e2` that
+     (1) evaluates e1, (2) takes `|- e1 = v1` and evaluates e2.
+     The benefit of using this continuation-passing style over repetitively
+     applying rewriting rules is its speed.
+     To understand the data structure of HOL Light's terms, you will want
+     to disable the term printer in OCaml REPL via
+     "#remove_printer pp_print_qterm;;". This can be enabled by
+     "#disable_printer pp_print_qterm;;" again. *)
```

Comment:
I left comments that explain functions based on my understanding. Please feel free to comment about the descriptions.